tests/domains_test.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+import scipp as sc
+from scipp.core.domains import find_domains
+import pytest
+
+x = sc.arange(dim='x', start=0, stop=10)
+
+
+def test_find_domains_fail_not_1d():
+    data = sc.ones(dims=['y', 'x'], shape=[2, 9], dtype='int64')
+    da = sc.DataArray(data=data, coords={'x': sc.concat([x, x], 'y')})
+    with pytest.raises(sc.DimensionError):
+        find_domains(da)
+
+
+def test_find_domains_fail_no_edges():
+    data = sc.ones(dims=['x'], shape=[9], dtype='int64')
+    da = sc.DataArray(data=data, coords={'x': x['x', 1:]})
+    with pytest.raises(sc.DimensionError):
+        find_domains(da)
+
+
+def test_find_domains_length_1():
+    data = sc.ones(dims=['x'], shape=[1], dtype='int64')
+    da = sc.DataArray(data=data, coords={'x': sc.array(dims=['x'], values=[0, 2])})
+    assert sc.identical(find_domains(da), da)
+
+
+def test_find_domains_length_2():
+    data = sc.ones(dims=['x'], shape=[2], dtype='int64')
+    da = sc.DataArray(data=data, coords={'x': sc.array(dims=['x'], values=[0, 1, 2])})
+    expected = sc.DataArray(data=sc.array(dims=['x'], values=[1], dtype='int64'),
+                            coords={'x': sc.array(dims=['x'], values=[0, 2])})
+    assert sc.identical(find_domains(da), expected)
+
+
+def test_find_domains_2_domains_length_2():
```

Comment:
```suggestion
def test_find_domains_2_domains_length_1():
```

Reply:
length was referring to the total input length, not the domains, but you are right that this is ambiguous.